src/main/scala/nodes/learning/LBFGS.scala

```diff
@@ -130,22 +130,27 @@ object LBFGSwithL2 extends Logging {
  * @param regParam L2 regularization
  * @param numIterations max number of iterations to run
  */
-class DenseLBFGSwithL2(
+class DenseLBFGSwithL2[T <: Vector[Double]](
     val gradient: Gradient.DenseGradient,
     val fitIntercept: Boolean = true,
     val numCorrections: Int = 10,
     val convergenceTol: Double = 1e-4,
     val numIterations: Int = 100,
     val regParam: Double = 0.0)
-  extends LabelEstimator[DenseVector[Double], DenseVector[Double], DenseVector[Double]] {
+  extends LabelEstimator[T, DenseVector[Double], DenseVector[Double]] {
+
+  def fit(data: RDD[T], labels: RDD[DenseVector[Double]]): LinearMapper[T] = {
+    val denseData = data.map {
+      case denseVec: DenseVector[Double] => denseVec
+      case vec => vec.toDenseVector
```

Comment:
My thinking was that if we type this estimator to work on any data that extends `Vector`, then it should be able to work on SparseVectors, converting to Dense if necessary. There are also cases where if the sparse vector has a high density, it's better to just make it fully dense anyway rather than trying to use sparse operations.